### ISSUE TITLE
TNSRecorder Error Domain=NSOSStatusErrorDomain Code=1718449215

### ISSUE THREAD

**Issue by dotnetdreamer** [nstudio/nativescript-audio#100]
I have added recorder functionality but i am getting following error:
TNSRecorder Error Domain=NSOSStatusErrorDomain Code=1718449215 "(null)"

Here is my code:
```
recordAudio(): Promise<TNSRecorder> {
        let recorder = new TNSRecorder();
        const uniqueId = this.getUniqueId();

        return new Promise(async (resolve, reject) => {
            if(!TNSRecorder.CAN_RECORD()) {
                reject('This device cannot record audio');
                return;
            }
            try {
                let audioFolder = fs.knownFolders.currentApp().getFolder("recordings");
                const fullPath = `${audioFolder.path}/${uniqueId}.mp3`;

                const recorderOptions: AudioRecorderOptions = {
                    filename: fullPath,
                    format: ".mp3",
                    infoCallback: () => {
                        console.log('infoCallback');
                    },
                    errorCallback: () => {
                        reject('Could not record audio');
                    }
                };
                recorder.debug = true;
                recorder.start(recorderOptions).then((res) => {
                    resolve(recorder);
                 }, (err) => {
                     reject(err);
                 });
            } catch (e) {
                reject(e);
            }
        });
    }
```

**Comment by MartinMitro** [comment]
@dotnetdreamer I had the same problem. Your file has to have extension .caf for the iOS, not .mp3. Look at the sample app. IMHO: this should be also in documentation

**Comment by dotnetdreamer** [comment]
@MartinMitro  in the 'format' property or fullpath ?

**Comment by dotnetdreamer** [comment]
@MartinMitro  i changed the extension as you said so, and its working fine now. Thanks man you are my hero 🥇

**Comment by dotnetdreamer** [comment]
@MartinMitro the only issue i have currently is when i upload the .caf file and then try to play .caf file, i am getting error:

`Error Domain = NSOSStatusErrorDomain Code203334207 "null"`
Any idea ?

**Comment by dotnetdreamer** [comment]
sorry. i realized there is `playFromUrl` method

**Comment by pajohns** [comment]
I was still getting this issue, but only as of recently after moving to webpack to build - which led me to believe that it was not copying the .mp3 file across - and it wasn't. Looking at the files that it was copying, there was no mp3 file included. To fix this I edited the `webpack.config.js` file and in the `plugins` section where it lists the files to include, I added the line `{ from: "audio/*.mp3" },` and this fixed it.
